src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -0,0 +1,706 @@
+        /// <returns><see langword="true" /> if the parse was successful, <see langword="false" /> otherwise.</returns>
+        public static bool TryParse(ReadOnlySpan<char> s, out Half result)
+        {
+            return TryParse(s, DefaultParseStyle, provider: null, out result);
+        }
+
+        /// <summary>
+        /// Tries to parse a <see cref="Half"/> from a <see cref="string"/> with the given <see cref="NumberStyles"/> and <see cref="IFormatProvider"/>.
+        /// </summary>
+        /// <param name="s">The input to be parsed.</param>
+        /// <param name="style">The <see cref="NumberStyles"/> used to parse the input.</param>
+        /// <param name="provider">A format provider. </param>
+        /// <param name="result">The equivalent <see cref="Half"/> value representing the input string if the parse was successful. If the input exceeds Half's range, a <see cref="Half.PositiveInfinity"/> or <see cref="Half.NegativeInfinity"/> is returned. If the parse was unsuccessful, a default <see cref="Half"/> value is returned.</param>
+        /// <returns><see langword="true" /> if the parse was successful, <see langword="false" /> otherwise.</returns>
+        public static bool TryParse(string s, NumberStyles style, IFormatProvider? provider, out Half result)
+        {
+            return TryParse(s.AsSpan(), style, provider, out result);
+        }
+
+        /// <summary>
+        /// Tries to parse a <see cref="Half"/> from a <see cref="ReadOnlySpan{Char}"/> with the given <see cref="NumberStyles"/> and <see cref="IFormatProvider"/>.
+        /// </summary>
+        /// <param name="s">The input to be parsed.</param>
+        /// <param name="style">The <see cref="NumberStyles"/> used to parse the input.</param>
+        /// <param name="provider">A format provider. </param>
+        /// <param name="result">The equivalent <see cref="Half"/> value representing the input string if the parse was successful. If the input exceeds Half's range, a <see cref="Half.PositiveInfinity"/> or <see cref="Half.NegativeInfinity"/> is returned. If the parse was unsuccessful, a default <see cref="Half"/> value is returned.</param>
+        /// <returns><see langword="true" /> if the parse was successful, <see langword="false" /> otherwise.</returns>
+        public static bool TryParse(ReadOnlySpan<char> s, NumberStyles style, IFormatProvider? provider, out Half result)
+        {
+            bool ret = false;
+            if (float.TryParse(s, style, provider, out float floatResult))
+            {
+                result = (Half)floatResult;
+                ret = true;
+            }
+            else
+            {
+                result = default(Half);
+            }
+            return ret;
```

Comment:
```suggestion
            bool ret = float.TryParse(s, style, provider, out float floatResult))
            
            result = ret ? (Half)floatResult : default(Half);

            return ret;
```
? assume it makes the same code

Reply:
Rather than calling into `float.TryParse` we should add a `Number.TryParseHalf` method. The method would ultimately be the same as https://source.dot.net/#System.Private.CoreLib/Number.Parsing.cs,1792 but it would return `Half.PositiveInfinity`, `Half.NegativeInfinity`, `Half.NaN`, and use a new `HalfNumberBufferLength` constant (this should be 21 if I did my math correctly).

There would then also be a new `NumberToHalf` method which likewise mirrors https://source.dot.net/#System.Private.CoreLib/Number.Parsing.cs,2002, but passes in `FloatingPointInfo.Half`. The rest of the logic should already be correct and should not require any touchups.

Similar should be true for the formatting code paths.

Reply:
Hmm, a bit longer?
https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLRjgOwMYwJYD2mqADgXpjBAsgDQwhTIC2APgAIBMAjALABQ7AAwACdtwB0AcQA2BYFBl4AXrEKYA3APYBmMZxEBhEQG8BIi5bF7xANhHACBGSIAqCAJ4AFKDQgAKACVoABMAeUwZDwBlEihMAB5sAAtfAD4ROhEAOThmYGpomA8ZFEzi0toRAEkAMQIEZlgvBAIANzwQ6hESVo6uhCqCOBgRAAlFMBEkZDgZGABKcytLM34VjYcnFyRRgF4RMEVkCC11zas8Kf8wOVgJd29fE/8s5AqIKt72zuohkcOd1GtwIsGCs3mCyW5wuFjWsM2Mzm+xE/gmMjACxBYJQyLOCI2uxEBxgCDgp2WCIAvpTYRAZCdaRd4QSrEj5sSRF0jsi0ZMFviCTSYbD2AB2aYQGBnYVWJkbXRibj2RzONyeHx+ThBUIRKKxeJJVIIDJZXL5QofZDlEqfGr1RrNPq/BA9Z0Df6jdFTdmLeVw/0rVU7KWc7EwB4a54BN4fL7uv4iYbAoGA0EwcHIgWBuUii6+zlEgD8qO9WKBmY5IC5EB58z5GOzec24sl0sDssstMV7zJuHGkxZFiZirsIggAA8SEpsHhRgQSNRYA0Bxibqm2opyQsTGIJZgIAB3VdgfwCzsWYVUoA==